react-hot-button-component/my-app/src/App.js

```diff
@@ -0,0 +1,47 @@
+import CustomButton from './CustomButton';
+import { useState } from 'react';
+import './App.css';
+
+function App() {
+  const [counter, setCounter] = useState(0);
+
+  function handleClickEvent() {
+    setCounter(counter + 1);
+  }
+
+  let color =
+    counter === 3
+      ? 'rgb(53, 28, 117)'
+      : counter === 6
+      ? 'rgb(103,78,167)'
+      : counter === 9
+      ? 'rgb(224, 102, 102)'
+      : counter === 12
+      ? 'rgb(246,178,107)'
+      : counter === 15
+      ? 'rgb(255,255,0)'
+      : counter === 18
+      ? 'rgb(255, 255, 255)'
+      : '(255, 255, 255)';
+
+  let textColor = 'black';
+  if (counter >= 12) {
+    textColor = 'black';
+  } else if (counter >= 3) {
+    textColor = 'white';
+  }
```

Comment:
I would move all of this logic to the `CustomButton` component itself. In this case, since we're only rendering one instance of this component, and our `counter` state variable isn't going to impact any other components, it would be better to keep the logic within `CustomButton`.